### PR TITLE
ISSD-3188 [PRM] Show required field validation on PRM forms

### PR DESCRIPTION
**Pull request by eduardo-diniz**

### CODE REVIEW

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/ActivityClaimPanel.tsx [hunk 1]
```diff
@@ -64,12 +68,27 @@ const activityClaimStatusClassName = {
...
+	errors,
 	hasPermissionEditClaimActivity,
+	isButtonClicked,
+	isDraft,
 	overallCampaignDescription,
 	setFieldValue,
 }: IProps & Pick<FormikContextType<MDFClaim>, 'setFieldValue'>) => {
 	const [expanded, setExpanded] = useState<boolean>(!activity.selected);
 
+	const isNotBudgetSelected = Array.isArray(errors?.activities)
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/ActivityClaimPanel.tsx:80]: would it be possible to invert this variable naming and initialization? like make it named `isBudgetSelected`, and when using it negate it like `!isBudgetSelected`. we don't normally add the negation to the name like `isNot`

Comment (eduardo-diniz) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/ActivityClaimPanel.tsx]: Ok, sounds good to me!
Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx [hunk 2]
```diff
@@ -134,12 +161,12 @@ const Goals = ({
...
-								name="liferayBusinessSalesGoalsOther"
-								required
-							/>
-						)}
+								<PRMFormik.Field
+									component={PRMForm.InputText}
+									name="liferayBusinessSalesGoalsOther"
+									required
+								/>
+							)}
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx:169]: is this extra indentation needed

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx [hunk 3]
```diff
@@ -124,7 +151,7 @@ const Goals = ({
 						component={PRMForm.CheckboxGroup}
 						items={
 							fieldEntries[
-								LiferayPicklistName.LIFERAY_BUSINESS_SALES_GOALS
+							LiferayPicklistName.LIFERAY_BUSINESS_SALES_GOALS
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx:154]: this should still be indented

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx [hunk 4]
```diff
@@ -84,6 +86,31 @@ const Goals = ({
...
+			behavior: (!isValid ? 'instant' : 'smooth') as ScrollBehavior,
+			top: 0,
+		});
+
+		onContinue?.(formikHelpers, StepType.ACTIVITIES);
+	};
+
+	const hasError = !isValid && !isObjectEmpty(goalsErrors);
+
+	const isEdiForm = Boolean(values.id);
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx:107]: typo? `Edi` is not a word

Comment (eduardo-diniz) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestForm/steps/Goals/Goals.tsx:107]: Done! Thanks!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useSetTouchedOnForms.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {FormikErrors, FormikHelpers, setNestedObjectValues} from 'formik';
```
Comment (ryanschuhler) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useSetTouchedOnForms.ts:6]: Do we have any in house standard for form/error handling? Feels odd to be importing an external library for error handling

**Review by thaynam — CHANGES_REQUESTED** [review]

**Comment by thaynam** [comment]
@eduardo-diniz your pr has unrelated commits, pls resend

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/MDFClaimPage.tsx [hunk 6]
```diff
@@ -151,6 +159,18 @@ const MDFClaimPage = ({
...
 
+		const handleOnClick = () => {
+			setIsButtonClicked(true);
+			window.scrollTo({
+				behavior: (isValid ? 'instant' : 'smooth') as ScrollBehavior,
+				top: 0,
+			});
+		};
+
+		const isDisabledButton =
```
Comment (thaynam) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/MDFClaimPage.tsx:170]:
```suggestion
		const isButtonDisabled =
```

**Comment by thaynam** [comment]
we need a similar touched behavior in the Type of Activity and Tactic fields (Activity) for editing mdf requests

**Review by thaynam — CHANGES_REQUESTED** [review]

**Comment by thaynam** [comment]
[resent](https://github.com/liferay-customer-delivery/liferay-portal/pull/102)

**Pull request closed without merging**